### DISCUSSION TITLE
SendGrid SMTP reply-to header

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#5160] [Q&A]
Has anyone been able to successfully leverage a reply-to header when sending custom emails on PocketBase using SendGrid SMTP?

I've tried both setting the `Reply-To` header directly and also the `X-SMTPAPI` header, individually and simultaneously, like the example below:
 
```
	message := &mailer.Message{
		From: mail.Address{
			Name:    app.Config().Mail.RewardsName,
			Address: app.Config().Mail.RewardsSender,
		},
		To:      []mail.Address{{Address: member.Email}},
		Subject: subject,
		HTML:    body,
		Headers: map[string]string{
			// see https://www.twilio.com/docs/sendgrid/for-developers/sending-email/building-an-x-smtpapi-header
			"X-SMTPAPI": fmt.Sprintf(`{"reply_to":"%s"}`, account.Email),
			"Reply-To":  account.Email,
		},
	}
```

Either or, the email (which is sent successfully) simply does not include the reply-to header and I can't figure out why.

I will be contacting SendGrid support, but just wondering if anyone has come across a similar requirement here.

Thanks,

**Comment by ganigeorgiev** [comment]
>  the email (which is sent successfully) simply does not include the reply-to header 

Try to explore the raw email headers to verify that your 2 headers are added (for emails sent to Gmail you can do this by clicking on the _3-vertical dots > Show original_).

From my [last test couple months ago](https://github.com/pocketbase/pocketbase/discussions/4595#discussioncomment-8848877) the `Reply-To` header is added correctly but it would depend on the receiver's mail client whether it will acknowledge it or ignore it.

**Reply by pnmcosta** [reply]
Yah, I checked on Gmail and Proton, both are empty.

I really think it's a SendGrid setting/restriction of some sort that is stripping/ignoring it, or I'm not using the `X-SMTPAPI` header properly,  cause the email I'm using on the header is valid and operational (has MX headers and validates).

I'll report back my findings as soon as I figure it out.
